SouvlakMVP/SouvlakMVP/Dijkstra.cs

```diff
@@ -6,65 +6,160 @@
+        }
+
+        return pathsAndCosts;
+    }
+
+    /// <summary>Finds the shortest paths between all combinations of vertices pairs (RoundRobin)</summary>
+    /// <param name="graph">The graph to search.</param>
+    /// <param name="vertices">The list of vertices.</param>
+    /// <returns>A dictionary containing the shortest path and weight for every vertices pair.</returns>
+    public static Dictionary<HashSet<indexT>, (List<indexT>, edgeWeightT)> FindShortestPath(Graph graph, List<indexT> vertices)
+    {
+        List<(indexT, indexT)> verticesPairs = GetAllPairs(vertices);
+        List<indexT> uniqueStartVertices = new List<indexT>();
+
+        // Fill in the list of unique staring vertices
+        foreach ((indexT, indexT) pair in verticesPairs) 
+        {
+            if (!uniqueStartVertices.Contains(pair.Item1))
+            {
+                uniqueStartVertices.Add(pair.Item1);
+            }
+        }
+
+        Dictionary<HashSet<indexT>, (List<indexT>, edgeWeightT)> result = new Dictionary<HashSet<indexT>, (List<indexT>, edgeWeightT)>();
+
+        // Calculate paths and costs for every unique verices
+        foreach (indexT StartVertex in uniqueStartVertices)
+        {
+            Dictionary<indexT, (List<indexT>, edgeWeightT)> pathsAndCosts = FindShortestPath(graph, StartVertex);
+            foreach ((indexT, indexT) pair in verticesPairs.Where(pair => pair.Item1 == StartVertex))
+            {
+                (List<indexT>, edgeWeightT) pathAndCost = pathsAndCosts[pair.Item2];
+                HashSet<indexT> pairSet = new HashSet<indexT>() { pair.Item1, pair.Item2 };
+                result.Add(pairSet, pathAndCost);
+            }
+        }
+
+        return result;
+    }
+
+    /// <summary>Calculate preceding vertices list and list of minimal costs using Dijkstra's algorithm.</summary>
+    /// <param name="graph">The graph to search.</param>
+    /// <param name="startVertex">The vertex to start the search from.</param>
+    /// <param name="endVertex">The destination vertex of the search.</param>
+    /// <returns>A tuple containing 2 lists: preceding vertices; minimal costs to vertices.</returns>
+    public static (indexT?[], edgeWeightT[]) CalcClassicDijkstra(Graph graph, indexT startVertex, indexT? endVertex = null)
+    {
+        int verticesN = graph.GetVertexCount();
 
         // Array containing the minimum costs to reach each vertex from the starting vertex
-        distanceT[] minCostToVertex = new distanceT[verticesN];
+        edgeWeightT[] minCostToVertex = new edgeWeightT[verticesN];
         // Array containing the preceding vertices on the path from the starting vertex
         indexT?[] precedingVertices = new indexT?[verticesN];
 
         // Working lists of vertices
-        List<indexT> verticesToProcess = new List<indexT>();
-        List<indexT> processedVertices = new List<indexT>();
+        List<indexT> unvisitedVertices = new List<indexT>();
```

Comment:
For `unvisitedVertices` I'd suggest using [PriorityQueue<TElement, TPriority>](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.priorityqueue-2?view=net-7.0) instead of looking for _nearest_ vertex manually.
```suggestion
        PriorityQueue<indexT, minCostToVertex> unvisitedVertices = new List<indexT>();
```

Reply:
But well... let's do it later -> moving it to issue #22